octokit.gemspec

```
@@ -5,6 +5,7 @@ $LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
 require 'octokit/version'
 
 Gem::Specification.new do |spec|
+  spec.add_dependency 'base64'
```

```suggestion
  spec.add_dependency 'base64' if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('3.3')
```

What are thoughts about approaching it like this? I know that it shouldn't matter but I would like to reduce the surface area of the impact.  Given that pre v3.3 can get`base64` from stdlib without the warn I think this approach should be ok.

Let me know your thoughts when you get the chance!  Thanks.

---

It's not working. I'm member of Ruby core team and RubyGems team.

RubyGems couldn't handle condition of Ruby version for dependency. RubyGems only support `required_ruby_version` and `required_rubygems_version` for gemspec, not dependency.

---

Thanks for the clarity!  I've merged it as is and will ship it once I get a few more changesets in.